### PR TITLE
Add protocol for puma datadog control_url integration

### PR DESCRIPTION
**Pull request by kutysam**
Without this, DD agent will throw errors

```
      Instance ID: puma:9e09e7ef4ea3fdef [ERROR]
      Configuration Source: container:containerd://bd74cc0c520dfdb9e4431bf6f843782fed5018d70da6f4004a7fdce8bf8e62fb
      Total Runs: 2
      Metric Samples: Last Run: 0, Total: 0
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 1, Total: 2
      Average Execution Time : 4ms
      Last Execution Date : 2023-06-14 18:27:44 UTC (1686767264000)
      Last Successful Execution Date : Never
      Error: 'NoneType' object has no attribute 'endswith'
      Traceback (most recent call last):
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/utils/http.py", line 608, in should_bypass_proxy
          ipaddress = ip_address(ensure_unicode(parsed_uri))
        File "/opt/datadog-agent/embedded/lib/python3.8/ipaddress.py", line 53, in ip_address
          raise ValueError('%r does not appear to be an IPv4 or IPv6 address' %
      ValueError: None does not appear to be an IPv4 or IPv6 address

      During handling of the above exception, another exception occurred:

      Traceback (most recent call last):
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/checks/base.py", line 1135, in run
          self.check(instance)
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/puma/puma.py", line 24, in check
          response, content_type, version = self._perform_service_check(instance, control_url)
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/puma/puma.py", line 72, in _perform_service_check
          request = self._perform_request(url)
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/puma/puma.py", line 83, in _perform_request
          request = self.http.get(url)
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/utils/http.py", line 355, in get
          return self._request('get', url, options)
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/utils/http.py", line 379, in _request
          if self.no_proxy_uris and should_bypass_proxy(url, self.no_proxy_uris):
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/utils/http.py", line 624, in should_bypass_proxy
          if no_proxy_uri == parsed_uri or parsed_uri.endswith(dot_no_proxy_uri):
      AttributeError: 'NoneType' object has no attribute 'endswith'
```

### CODE REVIEW

**Comment by kutysam** [comment]
@fanny-jiang could you review this please.

**Comment by kutysam** [comment]
Anyone?

File: puma/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # CHANGELOG - puma
 
+## 1.3.0 / 2023-06-15
+
+* [Fixed] Update conf.yaml to refect http prefix. See [#1887](https://github.com/DataDog/integrations-extras/pull/1887).
+
```
Comment (alopezz) [inline puma/CHANGELOG.md:6]: We've changed the changelog format (see other entries):

```suggestion
## 1.2.1 / 2023-10-05

***Fixed***:

* Update conf.yaml to refect http prefix ([#1887](https://github.com/DataDog/integrations-extras/pull/1887))

```

File: puma/datadog_checks/puma/__about__.py [hunk 2]
```diff
@@ -1 +1 @@
-__version__ = '1.2.0'
+__version__ = '1.3.0'
```
Comment (alopezz) [inline puma/datadog_checks/puma/__about__.py:1]:
```suggestion
__version__ = '1.2.1'
```

**Review by alopezz — CHANGES_REQUESTED** [review]
Other than nits around the version segment to bump and changelog formatting, this looks good. Thanks!

File: puma/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # CHANGELOG - puma
 
+## 1.2.1 / 2023-10-05
+
+***Fixed***:
+
+* Update conf.yaml to refect http prefix ([#1887](https://github.com/DataDog/integrations-extras/pull/1887))
```
Comment (alopezz) [inline puma/CHANGELOG.md:7]: Just saw this typo that I missed on my first pass.

```suggestion
* Update conf.yaml to reflect http prefix ([#1887](https://github.com/DataDog/integrations-extras/pull/1887))
```

Comment (kutysam) [reply puma/CHANGELOG.md:7]: mb i too didn't realize this.

**Review by alopezz — APPROVED** [review]
Thanks!

**Pull request merged**